src/awkward/_v2/_connect/cling.py

```diff
@@ -134,11 +142,22 @@ class ArrayView {
       return start_ == stop_;
     }
 
+    PyObject* lookup() {
+        return lookup_;
+    }
+
+    PyObject* generator() {
+        return generator_;
+    }
+
   protected:
     ssize_t start_;
     ssize_t stop_;
     ssize_t which_;
     ssize_t* ptrs_;
+    PyObject* lookup_;
+    PyObject* generator_;
+
```

Comment:
@jpivarski - I'm not sure how to avoid adding both lookup and generator pointers. The generator needs a Form  of a lookup to convert it to a layout.

Reply:
As I (re-)learned on Zoom last week, a Generator is a subclass of a Lookup, so it could be one and the same object. That's not how it was implemented in Python: the `generator` is made first, from a Form, and the `lookup` is created afterward, from an Array. (In Numba, there's only one object.)

But it doesn't have to stay that way: the Python `generator` is made first, followed by a `lookup`. You could attach the original `generator` to the `lookup` as an attribute:

```python
lookup._generator = generator
```

and then only pass around `PyObject* lookup`. When you need to reconstitute a layout from a Lookup, you can assume that the `lookup._generator` exists, to get the Form.

Alternatively, you could attach only what you need:

```python
lookup._form = generator.form
```

Reply:
I've added a `generator` as a `Lookup` instance attribute because its other attributes do not have an `_` in their names.